Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Content.Server.Utility;
+using Content.Shared.Chemistry;
+using Content.Shared.GameObjects.Components.Nutrition;
+using Content.Shared.Interfaces;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Localization;
+using Robust.Shared.ViewVariables;
+using Serilog;
+
+namespace Content.Server.GameObjects.Components.Nutrition
+{
+    [RegisterComponent]
+    public class CreamPiedComponent : SharedCreamPiedComponent, IReagentReaction, IThrowCollide
+    {
+        private bool _creamPied;
+
+        [ViewVariables]
+        public bool CreamPied
+        {
+            get => _creamPied;
+            private set
+            {
+                _creamPied = value;
+                if (Owner.TryGetComponent(out AppearanceComponent appearance))
+                {
+                    appearance.SetData(CreamPiedVisuals.Creamed, CreamPied);
+                }
+            }
+        }
+
+        public void Wash()
+        {
+            if(CreamPied)
+                CreamPied = false;
+        }
+
+        public ReagentUnit ReagentReactTouch(ReagentPrototype reagent, ReagentUnit volume)
+        {
+            switch (reagent.ID)
+            {
+                case "chem.SpaceCleaner":
+                case "chem.H2O":
+                    Wash();
+                    break;
+            }
+
+            return ReagentUnit.Zero;
+        }
+
+        public void ThrowCollide(ThrowCollideEventArgs eventArgs)
+        {
+            if (eventArgs.Target != Owner || !eventArgs.Thrown.TryGetComponent(out CreamPieComponent creamPie) || CreamPied) return;
+
+            CreamPied = true;
+            Owner.PopupMessage(Loc.GetString("You have been creamed by {0:theName}!", eventArgs.User));
```

Comment:
this message seems... off lol.

Reply:
Also copying SS13 in this regard :cold_sweat: `H.visible_message("<span class='warning'>[H] is creamed by [src]!</span>", "<span class='userdanger'>You've been creamed by [src]!</span>")`
Any suggestions to make it better?

Reply:
Your message doesn't match SS13's.

Yours is "You have been creamed by bob!", SS13's is "You have been creamed by the cream pie!"

Reply:
oh no, `eventArgs.User` is the person who creamed you, while `eventArgs.Thrown` is the cream pie itself in this case

Reply:
Yes, and you're using user. SS13 uses the pie.

Reply:
oh

Reply:
I blame BYOND's shitty syntax

Reply:
Not really a syntax issue, you just misinterpreted byond's version of `this` :P